include/diagnostics/raftery.hpp

```diff
@@ -0,0 +1,116 @@
+// VolEsti (volume computation and sampling library)
+
+// Copyright (c) 2012-2020 Vissarion Fisikopoulos
+// Copyright (c) 2018-2020 Apostolos Chalkis
+
+//Contributed and/or modified by Alexandros Manochis, as part of Google Summer of Code 2020 program.
+
+//Licensed under GNU LGPL.3, see LICENCE file
+
+//Based on Matlab version of coda package in http://www.spatial-econometrics.com/gibbs/
+
+#ifndef RAFTERY_HPP
+#define RAFTERY_HPP
+
+template <typename NT>
+NT fix(NT x)
+{
+    if (x > 0.0) return std::floor(x);
+    return std::ceil(x);
+}
+
+#include "raftery_subroutines/empquant.hpp"
+#include "raftery_subroutines/indtest.hpp"
+#include "raftery_subroutines/mctest.hpp"
+#include "raftery_subroutines/mcest.hpp"
+#include "raftery_subroutines/thin.hpp"
+#include "raftery_subroutines/ppnd.hpp"
+
+
+template <typename VT, typename MT, typename NT>
+std::pair<MT,VT> perform_raftery(MT const& samples, NT const& q, NT const& r, NT const& s)
```

Comment:
another different interface: returns a pair and there are some q,r,s parameters that is unclear to understand

Reply:
I have added comments to provide information about raftery diagnostic